flang/lib/Optimizer/Dialect/FIRType.cpp

```diff
@@ -360,6 +360,17 @@ bool isRecordWithAllocatableMember(mlir::Type ty) {
   return false;
 }
 
+bool isRecordWithDescriptorMember(mlir::Type ty) {
+  if (auto recTy = ty.dyn_cast<fir::RecordType>())
+    for (auto [field, memTy] : recTy.getTypeList()) {
+      if (mlir::isa<fir::BaseBoxType>(memTy))
+        return true;
+      if (memTy.isa<fir::RecordType>() && isRecordWithDescriptorMember(memTy))
```

Comment:
Could also be a fir.array<fir.type<... {fir.box}>>, so you should likely unwrap any SequenceType here.

Reply:
Good catch! Thank you, Jean!